File: practica4 para revisar/practica4.java
package practica4;

public class practica4 {
    
    public static double calcular(double radio) {
        return Math.PI * radio * radio;
    }

  
    public static double calcular(double base, double altura) {
        return base * altura;
    }

   
    public static double calcular(double base, double altura, boolean esTriangulo) {
        return esTriangulo ? (base * altura) / 2 : base * altura;
    }

   
    public static double calcular(double baseMayor, double baseMenor, double altura) {
        return ((baseMayor + baseMenor) * altura) / 2;
    }

   
    public static double calcular(double lado, double apotema, int lados) {
        return (lados * lado * apotema) / 2;
    }

    public static void main(String[] args) {
        System.out.println("Área del círculo: " + calcular(5));  // 78.54
        System.out.println("Área del rectángulo: " + calcular(4, 6));  // 24
        System.out.println("Área del triángulo: " + calcular(4, 6, true));  // 12
        System.out.println("Área del trapecio: " + calcular(6, 4, 5));  // 25
        System.out.println("Área del pentágono: " + calcular(5, 3.5, 5));  // 43.75
    }
}

File: practica4 para revisar/practica4.py
import math

class Area:
    @staticmethod
    def calcular(radio=None, base=None, altura=None, base_mayor=None, base_menor=None, lados=None, apotema=None):
        """Calcula el área de distintas figuras geométricas."""
        
        if radio is not None and base is None and altura is None:
            
            return math.pi * radio ** 2
        
        elif base is not None and altura is not None and base_mayor is None and base_menor is None:
            
            return (base * altura) / 2 if lados is not None and lados == 3 else base * altura

        elif base_mayor is not None and base_menor is not None and altura is not None:
            
            return ((base_mayor + base_menor) * altura) / 2
        
        elif lados is not None and apotema is not None and base is not None:
            
            return (lados * base * apotema) / 2

        else:
            return "Parámetros inválidos"

# Pruebas
print("Área del círculo:", Area.calcular(radio=5))  # 78.54
print("Área del rectángulo:", Area.calcular(base=4, altura=6))  # 24
print("Área del triángulo:", Area.calcular(base=4, altura=6, lados=3))  # 12
print("Área del trapecio:", Area.calcular(base_mayor=6, base_menor=4, altura=5))  # 25
print("Área del pentágono:", Area.calcular(base=5, apotema=3.5, lados=5))  # 43.75

